krabcake/kc_main.c

```diff
@@ -153,10 +153,11 @@ static Bool kc_handle_client_request ( ThreadId tid, UWord* arg, UWord* ret )
 
    switch(arg[0]) {
    case VG_USERREQ__BORROW_MUT: {
-      VG_(dmsg)("kc_handle_client_request, handle BORROW_MUT %llx %llx\n",
-                arg[1], arg[2]);
+      VG_(dmsg)("kc_handle_client_request, handle BORROW_MUT %llx (<- return value) %llx %llx %llx %llx\n",
+                arg[1], arg[2], arg[3], arg[4], arg[5]);
       *ret = arg[1];
       handled = True;
+      break;
```

Comment:
can't believe i overlooked this. its going to be good to get all this code moved over to the rs staticlib once I get that working. :)